services/venice-router/src/main/java/com/linkedin/venice/router/httpclient/HttpClient5StorageNodeClient.java

```diff
@@ -128,12 +130,23 @@ public int getStatusCode() {
     }
 
     @Override
-    public ByteBuf getContentInByteBuf() throws IOException {
+    public ByteBuf getContentInByteBuf(boolean usePooledBuffer, ByteBufAllocator allocator) throws IOException {
```

Comment:
I feel this function signature is a bit strange to me. If `allocator` is not null, then `usePooledBuffer` must be true, correct? Is there a case we pass a allocator but we don't use it?
I prefer to define two methods
```
public ByteBuf getContentInByteBuf(ByteBufAllocator allocator)
public ByteBuf getContentInByteBuf() {
    getContentInByteBuf(null)
}
```